### ISSUE TITLE
Bug: Unable to import "../../deps/petal_components/assets/default.css"; in app.css file

### ISSUE THREAD

**Issue by gsperka** [petalframework/petal_components#154]
Thanks for the Petal updates! Loved implementing this into our codebase. I was on version `0.9` for a while and noticed the major update broke a lot of my original styling. I followed https://petal.build/components and am having an issue here:
```
@import "tailwindcss/base"; 
@import "../../deps/petal_components/assets/default.css"; 
@import "tailwindcss/components"; 
@import "tailwindcss/utilities";
```

The problem is that `@import "../../deps/petal_components/assets/default.css";` is simply not being imported / built. 

Is this an esbuild problem? This problem is documented here: [elixir - add CSS files to Phoenix - Stack Overflow](https://stackoverflow.com/a/71591398/3007294) but I followed these instructions and still no luck. 

Any ideas? I also imported the css file in the app.js file (which still doesn’t make sense to me since it’s a css file and I am coming from a webpack background…) but that didn’t work either…

Is using a layer the only work around? (https://tailwindcss.com/docs/adding-custom-styles#using-css-and-layer)

**Comment by mplatts** [comment]
Hmm potential issues:
- is your css file in the usual spot? `assets/css/app.css`?
- have you updated to v1 and run `mix deps.get`?
- can you browse to `deps/petal_comopnents/assets/default.css` in your code editor and see it there?

**Comment by mplatts** [comment]
Also have you done this step:
<img width="642" alt="CleanShot 2023-03-02 at 17 35 42@2x" src="https://user-images.githubusercontent.com/184383/222349901-deae71cd-58b6-4c46-a6ec-52f5eb0b9375.png">

**Comment by Awlexus** [comment]
Hi, I'm having the same issue. 
* Yes, the CSS file is in `assets/css/app.css`
* Yes, the version is updated to `1.0.4`
* I can access `deps/petal_components/assets/default.css` from my code editor and calling `cat ../../deps/petal_components/assets/default.css` from `assets/css/` works too

**Comment by gsperka** [comment]
> Hmm potential issues:
> 
> * is your css file in the usual spot? `assets/css/app.css`?
> * have you updated to v1 and run `mix deps.get`?
> * can you browse to `deps/petal_comopnents/assets/default.css` in your code editor and see it there?

Yeah - I did all of that as well. 

I even went as far as to just copy and pasting the `default.css` file directly (all 1000+ lines) straight into the `app.css` file to see what happens and I got a mixed bag of results. When you inspect the assets loaded on page load, they're there, but certain styles are not being applied at all. 

For example, `pc-container` is applied but `pc-container--xl` is not. I tried the `@layer` approach and had the same result as well.

**Comment by mplatts** [comment]
Sounds like a tailwind config issue. Hard for me to debug without looking at it.. if you don’t mind me seeing your app, could you add me to the github repo?

**Comment by gsperka** [comment]
> Sounds like a tailwind config issue. Hard for me to debug without looking at it.. if you don’t mind me seeing your app, could you add me to the github repo?

HI @mplatts  - thanks for offering to help. Unfortunately the repo belongs to the company I work for and unable to provide access to. I can, however, provide you with the relevant files that I believe are all involved: 

**dev.exs**:

```
config :COMPANYNAME, COMPANYNAME.Endpoint,
  # Binding to loopback ipv4 address prevents access from other machines.
  # Change to `ip: {0, 0, 0, 0}` to allow access from other machines.
  http: [ip: {127, 0, 0, 1}, port: 4000],
  check_origin: false,
  code_reloader: true,
  debug_errors: true,
  secret_key_base: "xyz",
  watchers: [
    tailwind: {Tailwind, :install_and_run, [:default, ~w(--watch)]},
    # Start the esbuild watcher by calling Esbuild.install_and_run(:default, args)
    esbuild: {Esbuild, :install_and_run, [:default, ~w(--sourcemap=inline --watch)]}
  ]
```

**config.exs**:

```
# Configure esbuild (the version is required)
config :esbuild,
  version: "0.17.10",
  default: [
    args:
      ~w(js/app.js --bundle --target=es2017 --outdir=../priv/static/assets --external:/fonts/* --external:/images/*),
    cd: Path.expand("../assets", __DIR__),
    env: %{"NODE_PATH" => Path.expand("../deps", __DIR__)}
  ]

config :tailwind,
  version: "3.0.7",
  default: [
    args: ~w(
            --config=tailwind.config.js
            --input=css/app.css
            --output=../priv/static/assets/app.css
            ),
    cd: Path.expand("../assets", __DIR__)
  ]

config :petal_components,
       :error_translator_function,
       {COMPANYNAME.ErrorHelpers, :translate_error}
```

**tailwind.config.js**

```
const colors = require("tailwindcss/colors");

module.exports = {
  content: [
    "../lib/*_web.ex",
    "../lib/*_web/**/*.*ex",
    "./js/**/*.js",
    "../deps/petal_components/**/*.*ex", // <-- ADD THIS LINE
  ],
  darkMode: "class",
  theme: {
    extend: {

      // ADD THESE COLORS (can pick different ones from here: https://tailwindcss.com/docs/customizing-colors)
      colors: {
        primary: colors.blue,
        secondary: colors.pink,
        success: colors.green,
        danger: colors.red,
        warning: colors.yellow,
        info: colors.sky,

        // Options: slate, gray, zinc, neutral, stone
        gray: colors.gray,
      },
    },
  },

  plugins: [
    require("@tailwindcss/forms"),
    plugin(({ addVariant }) =>
      addVariant("phx-no-feedback", [".phx-no-feedback&", ".phx-no-feedback &"])
    ),
    plugin(({ addVariant }) =>
      addVariant("phx-click-loading", [
        ".phx-click-loading&",
        ".phx-click-loading &",
      ])
    ),
    plugin(({ addVariant }) =>
      addVariant("phx-submit-loading", [
        ".phx-submit-loading&",
        ".phx-submit-loading &",
      ])
    ),
    plugin(({ addVariant }) =>
      addVariant("phx-change-loading", [
        ".phx-change-loading&",
        ".phx-change-loading &",
      ])
    ),
  ],
};
```

app.css (path: assets/css/app.css)

```
@import "tailwindcss/base";
@import "../../deps/petal_components/assets/default.css";
@import "tailwindcss/components";
@import "tailwindcss/utilities";

....
```

If you need any other file references, let me know. Thanks again

**Comment by mplatts** [comment]
Looks pretty normal to me. Weird. Could try upping the tailwind version: `3.2.0`.

Here is a working copy: https://github.com/petalframework/petal_boilerplate

**Comment by gsperka** [comment]
Thanks for the `https://github.com/petalframework/petal_boilerplate` reference. I did download the boilerplate codebase and can confirm that it works. I am not sure how this setup differs from mine but I will look into it and keep you posted.

**Comment by gsperka** [comment]
@mplatts  - thanks again for the boilerplate reference. I was able to get it working! It turns out I needed to: 

- Upgrade to Phoenix 1.7
- Upgrade Tailwind (via `mix tailwind.install `)
- Properly configure `tailwind.config.js` to account for some missing class (I have already pre-defined primary and secondary classes that didn't have a "50" key - value pair)

One suggestion would be to update the documentation on https://petal.build/components. This line is missing from the top of the `tailwind.config.js` file in the **Modify the tailwind.config.js** section:  

`const plugin = require("tailwindcss/plugin");`
